### DISCUSSION TITLE
bad request permission: helm-chart:read

### DISCUSSION THREAD

**Discussion by jm-nab** [goharbor/harbor#20093] [General]
When using the v2 robots api:

`/#/robot/CreateRobot`

With the following curl request:

```
curl -X 'POST' \
  'http://test.local/api/v2.0/robots' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -H 'X-Harbor-CSRF-Token: asdfasdf==' \
  -d '{
      "description": "The robot account for Test",
      "name": "test",
      "secret": "asdfasdf",
      "level": "project",
      "disable": false,
      "duration": 23110,
      "permissions": [
        {
          "kind": "project",
          "namespace": "test",
          "access": [
            {
              "action": "list",
              "resource": "repository",
              "effect": "allow"
            },
            {
              "action": "pull",
              "resource": "repository",
              "effect": "allow"
            },
            {
              "action": "read",
              "resource": "artifact",
              "effect": "allow"
            },
            {
              "action": "list",
              "resource": "artifact",
              "effect": "allow"
            },
            {
              "action": "list",
              "resource": "tag",
              "effect": "allow"
            },
            {
              "action": "read",
              "resource": "helm-chart",
              "effect": "allow"
            }
          ]
        }
      ]
    }'
```

I get the following error:

```
{
  "errors": [
    {
      "code": "BAD_REQUEST",
      "message": "bad request permission: helm-chart:read"
    }
  ]
}
```

What am I missing?

Related:

- https://github.com/goharbor/harbor/blob/d25f3556a94a687d58faeac1197dd5090b17ec29/api/v2.0/swagger.yaml#L7764
- https://github.com/goharbor/harbor/blob/d25f3556a94a687d58faeac1197dd5090b17ec29/src/server/v2.0/handler/robot.go#L135
- https://github.com/goharbor/harbor/blob/main/src/common/rbac/project/rbac_util.go#L27

https://github.com/goharbor/harbor/blob/d25f3556a94a687d58faeac1197dd5090b17ec29/src/server/v2.0/handler/robot.go#L306

**Comment by jm-nab** [comment]
Looking at the policies map, i dont even see a `helm-chart` resource?

https://github.com/goharbor/harbor/blob/main/src/common/rbac/const.go#L84

**Comment by Vad1mo** [comment]
I guess you are looking at the latest harbor but expecting old behavior of chart museum times. Charts are now in OCI. So there is no destination between an image and a chart in OCI.

**Reply by jm-nab** [reply]
indeed, i'll try to update the PR, will put in draft. i had found this, and was just coming to that conclusion as well: https://github.com/goharbor/harbor/commit/738fde7d3bd8f201711907a826630b3fe5d13e16
